### PR TITLE
ci(build-deploy): build python deb/rpm for autoinjection

### PR DESCRIPTION
**Pull request by randomanderson**
This PR adds support to build .deb and .rpm packages for dd-trace-py. This is needed to support auto injection initiatives.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) are followed.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.

### CODE REVIEW

**Review by emmettbutler — DISMISSED** [review]

File: .gitlab/build-deb-rpm.sh [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+mkdir -p dd-trace.dir/lib
+
+# Install known compatible pip as default version shipped in Ubuntu (20.0.2)
+# does not work.
+python3 -m pip install -U "pip>=22.0"
+python3 -m pip install packaging
+
+echo `pwd`
+
+../lib-injection/dl_wheels.py \
```
Comment (bkabrda) [inline .gitlab/build-deb-rpm.sh:21]: Please note that if this is building binary shared objects, you should carefully consider which distribution versions you want to support with the packages. For example, we support CentOS/RHEL 6 with the Agent, which has glibc 2.12. Based on what you're aiming to support with these packages, you might need to build on a very old builder to get the proper glibc compatibility.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -1,9 +1,12 @@
 stages:
+  - package
```
Comment (bkabrda) [inline .gitlab-ci.yml:2]: I'm not 100 % sure how this works with the included files, but the included `packaging.yml` also has a `release` stage which I think should be added here.

**Review by bkabrda — APPROVED** [review]
Approving with a note for your consideration

**Review by Kyle-Verhoog — APPROVED** [review]
looks good to me!

File: .gitlab/build-deb-rpm.sh (hunk 1 above)
Comment (randomanderson) [reply .gitlab/build-deb-rpm.sh:21]: In talking to the team, it's okay that the python package won't support all OSes that the agent supports.

**Review by brettlangdon — APPROVED** [review]

**Pull request merged**